### PR TITLE
Add consul plugin

### PR DESCRIPTION
**Pull request by bodymindarts**
Plugin to back up consul KV store.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,8 @@ plugins:
 	go $(BUILD_TYPE) ./plugin/mysql
 	go $(BUILD_TYPE) ./plugin/rabbitmq-broker
 	go $(BUILD_TYPE) ./plugin/scality
+	go $(BUILD_TYPE) ./plugin/consul
+
```
Comment (jhunt) [inline Makefile]: You're also going to need to add a command to the `release` target to get this into the BOSH release

File: plugin/consul/plugin.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValue("host")
+	if err != nil {
+		ansi.Printf("@R{\u2717 host      %s}\n", err)
+		fail = true
+	} else {
+		ansi.Printf("@G{\u2713 host}      @C{%s}\n", s)
```
Comment (jhunt) [inline plugin/consul/plugin.go]: What happens if the `host` config value is the empty string?

**Comment by jhunt** [comment]
Looks like this PR is still missing deps?  Travis builds are still failing.

**Comment by geofffranks** [comment]
\o/ tests pass in travis now

File: plugin/consul/plugin.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	"io"
+	"os"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/starkandwayne/goutils/ansi"
+	. "github.com/starkandwayne/shield/plugin"
+)
+
+var (
+	DefaultPort = "5432"
```
Comment (jhunt) [inline plugin/consul/plugin.go]: Is this constant used?

**Comment by bodymindarts** [comment]
Nah its probably hanging around from copy pasting of the pg plugin.

**Comment by jhunt** [comment]
Cool.  Remove that constant, and we can merge.

**Pull request merged**